core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -208,6 +209,10 @@ class TransactionCoordinator(brokerId: Int,
     if (transactionalId == null || transactionalId.isEmpty) {
       responseCallback(Errors.INVALID_REQUEST)
     } else {
+      // if there is any partitions unknown in the metadata cache, return immediately to client
+      if (partitions.exists(tp => !metadataCache.contains(tp.topic, tp.partition)))
+        responseCallback(Errors.UNKNOWN_TOPIC_OR_PARTITION)
```

Comment:
Mentioned offline, but leaving it here as well: We should send this as a per partition error only for the partitions which are missing.